app/react/kubernetes/applications/components/RedeployGitAppForm/RedeployGitAppForm.tsx

```diff
@@ -0,0 +1,362 @@
+        }
+        onSubmit={handleSubmit}
+      >
+        <RedeployGitAppInnerForm
+          stack={stack}
+          namespaceName={namespaceName}
+          webhookId={webhookId}
+        />
+      </Formik>
+    </FormSection>
+  );
+
+  async function handleSubmit(
+    values: KubeAppGitFormValues,
+    { resetForm }: FormikHelpers<KubeAppGitFormValues>
+  ) {
+    trackSubmit(values);
+
+    // save the new git credentials if the user has selected to save them
+    if (
+      values.authentication.SaveCredential &&
+      values.authentication.NewCredentialName &&
+      values.authentication.RepositoryPassword &&
+      values.authentication.RepositoryUsername
+    ) {
+      createGitCredentialMutation.mutate({
+        name: values.authentication.NewCredentialName,
+        username: values.authentication.RepositoryUsername,
+        password: values.authentication.RepositoryPassword,
+        userId: user.Id,
+      });
+    }
+
+    // update the kubernetes git stack
+    const { authentication } = values;
+    const updateKubeGitStack: UpdateKubeGitStackPayload = {
+      RepositoryAuthentication:
+        !!values.authentication.RepositoryAuthentication,
+      RepositoryReferenceName: values.repositoryReferenceName,
+      AutoUpdate: transformAutoUpdateViewModel(values.autoUpdate, webhookId),
+      TLSSkipVerify: values.tlsSkipVerify,
+      RepositoryGitCredentialID: authentication.RepositoryGitCredentialID,
+      RepositoryPassword: authentication.RepositoryPassword,
+      RepositoryUsername: authentication.RepositoryUsername,
+    };
+    await updateKubeGitStackMutation.mutateAsync(updateKubeGitStack, {
+      onSuccess: ({ data: newStack }) => {
+        const newInitialValues = getInitialValues(newStack);
+        notifySuccess('Success', 'Application saved successfully.');
+        resetForm({ values: newInitialValues });
+      },
+    });
+  }
+
+  function trackSubmit(values: KubeAppGitFormValues) {
+    trackEvent('kubernetes-application-edit', {
+      category: 'kubernetes',
+      metadata: {
+        'automatic-updates': automaticUpdatesLabel(
+          values.autoUpdate.RepositoryAutomaticUpdates,
+          values.autoUpdate.RepositoryMechanism
+        ),
+      },
+    });
+    function automaticUpdatesLabel(
+      repositoryAutomaticUpdates: boolean,
+      repositoryMechanism: AutoUpdateMechanism
+    ) {
+      switch (repositoryAutomaticUpdates && repositoryMechanism) {
+        case RepositoryMechanismTypes.INTERVAL:
+          return 'polling';
+        case RepositoryMechanismTypes.WEBHOOK:
+          return 'webhook';
+        default:
+          return 'off';
+      }
+    }
+  }
+}
+
+function RedeployGitAppInnerForm({
+  stack,
+  namespaceName,
+  webhookId,
+}: {
+  stack: Stack;
+  namespaceName: string;
+  webhookId: string;
+}) {
+  const router = useRouter();
+  const environmentId = useEnvironmentId();
+  const redeployKubeGitStackMutation = useRedeployKubeGitStackMutation(
```

Comment:
this won't pass through form validation. maybe move it to outer form component, and send button name when submitting? then trigger needed mutation per the name

Reply:
It doesn't need to pass through validation, the redeploy button is disabled when the form is dirty